openmls/src/group/mls_group/creation.rs

```diff
@@ -81,11 +82,64 @@ impl MlsGroup {
             own_kpbs: vec![],
             aad: vec![],
             resumption_secret_store,
-            active: true,
+            group_state: MlsGroupState::Operational,
             state_changed: InnerState::Changed,
-            pending_commit: None,
         };
 
         Ok(mls_group)
     }
+
+    /// Creates a new group by joining an existing group through an External
+    /// Init. The resulting [`MlsGroup`] instance starts off with a pending
+    /// commit (the external commit, which adds this client to the group).
+    /// Merging this commit is necessary for this [`MlsGroup`] instance to
+    /// function properly, as, for example, this client is not yet part of the
+    /// tree. As a result, it is not possible to clear the pending commit. If
+    /// the external commit was rejected due to an epoch change, the
+    /// [`MlsGroup`] instance has to be discarded and a new one has to be
+    /// created using this function based on the latest `ratchet_tree` and
+    /// public group state. For more information on the external init process,
+    /// please see Section 11.2.1 in the MLS specification.
+    pub fn new_from_external_init(
```

Comment:
We should rethink the naming in general. This sounds very technical and something like `join_by_external_commit` might be more expressive. But we can do that in another PR.

Reply:
You're right. I somehow thought that the canonical name for the process was "External Init", but the spec is not very clear on that. There is the notion of an "External Commit", which allows you to join (and which is why your suggested function name makes sense) and in the Key Schedule, there is "External Initialization". But you're right in that "join through an external commit" is probably the best way to go.